go/vt/vtgate/engine/insert.go

```diff
@@ -766,9 +766,12 @@ func (ins *Insert) getInsertShardedRoute(
 			if keyspaceIDs[index] != nil {
 				mids = append(mids, sqlparser.String(ins.Mid[index]))
 				for _, expr := range ins.Mid[index] {
-					if arg, ok := expr.(*sqlparser.Argument); ok {
-						shardBindVars[arg.Name] = bindVars[arg.Name]
-					}
+					_ = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+						if arg, ok := node.(*sqlparser.Argument); ok {
+							shardBindVars[arg.Name] = bindVars[arg.Name]
```

Comment:
shouldn't we check if the `bindVars` has the value or not? here we would not fail on missing arguments, we would just put in a null value

Reply:
It could also be a user error when using grpc for the case when the user did not provide the bind var.